assets/js/app/Panel.js

```diff
@@ -66,50 +49,107 @@ LocusZoom.Panel = function() {
     this.getBaseId = function(){
         return this.parent.id + "." + this.id;
     };
+
+    // Initialize the layout (should this happen in initialize()?)
+    this.initializeLayout();
     
     return this;
     
 };
 
+LocusZoom.Panel.prototype.initializeLayout = function(){
+
+    // Set panel dimensions, origin, and margin or fall back to default values
+    this.layout.width      = this.layout.width      || 0;
```

Comment:
This is a complex bit of logic to get right. Compounding factors:

* Layouts are treated as nested objects (e.g. the main Instance/Plot layout passes each panel layout into the Panel classes and those in turn pass the data layer layouts into each Data Layer)
* The way JavaScript handles references those layouts on nested objects are references to slices of the original layout stored on the Plot/Instance
* Layout values can be arbitrary types and may be valid as more than one type (good example would be the `color` value for a data layer... could be a string representation of a color or an object that describes the coloring function to use, associated data and parameters, etc.)

I agree that for the most flexibility and best user experience one should be able to define no layout at all at any level or partial layouts and have the gaps "filled in" by the default layout. To make that a reality I'm still leaning in the direction of a Layout class where we can hang generic versions of those functions to be applied at each nested layer. This could also be a place to have generic functions to re-consume a changed layout to see those changes live, import new partial or complete layouts to merge them with current layout definitions, export layouts as serializable objects, etc.

So what about for this branch? Well we're just starting down this layout path... implementing such a class would be a big refactor on top of another big refactor so belongs in a subsequent branch. For this branch we should probably just make the default patterns consistent enough among each other. I'll get this line updated to pull from the default layout.